joern-cli/frontends/x2cpg/src/main/scala/io/joern/x2cpg/datastructures/ProgramSummary.scala

```diff
@@ -0,0 +1,253 @@
+package io.joern.x2cpg.datastructures
+
+import scala.collection.mutable
+
+/** A heirarchical data-structure that stores the result of types and their respective members. These types can be
+  * sourced from pre-parsing the application, or pre-computed stubs of common libraries.
+  *
+  * The utility of this object is in assisting resolving shorthand types during AST creation.
+  *
+  * @tparam M
+  *   the method/function meta data class.
+  * @tparam F
+  *   the field/object property meta data class.
+  * @tparam T
+  *   the type/class meta data class.
+  */
+trait ProgramSummary[M <: MethodLike, F <: FieldLike, T <: TypeLike[M, F]] {
```

Comment:
It doesn't look like `M` and `F` are really needed here, are they? This might help to simplify thinkgs as well. Can you please check the same for other type parameters as well?

```suggestion
trait ProgramSummary[T <: TypeLike[_, _]] {
```